rcl/src/rcl/node_options.c

```diff
@@ -57,6 +57,8 @@ rcl_node_options_copy(
   if (NULL != options->arguments.impl) {
     rcl_ret_t ret = rcl_arguments_copy(&(options->arguments), &(options_out->arguments));
     return ret;
+  } else {
+    options_out->arguments.impl = NULL;
```

Comment:
Looking at this in more depth, we may actually want to just return an error if arguments.impl is not NULL. To avoid memory leaks, `rcl_arguments_copy` returns an error if options_out->arguments.impl is not NULL before copying to it. If `impl` had been allocated before setting it to NULL here, there would be a similar memory leak.

Reply:
You mean checking arguments.impl of the options_out?

Reply:
Yah, just like rcl_arguments_copy does here:
https://github.com/ros2/rcl/blob/master/rcl/src/rcl/arguments.c#L791

Reply:
I agree with you about the memory leak, I'll erase this check and add a different check at the beginning of the function to address this.